spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -324,4 +325,12 @@ public boolean caseSensitive() {
         .defaultValue(SQLConf.CASE_SENSITIVE().defaultValueString())
         .parse();
   }
+
+  public String branch() {
+    return confParser
+        .stringConf()
+        .option(SparkWriteOptions.BRANCH)
+        .defaultValue(SnapshotRef.MAIN_BRANCH)
```

Comment:
@rdblue @aokolnychyi @namrathamyske As discussed in the community sync, we'll remove the SQL write option from this PR and take that separately since for row level operations there's more fundamental changes that need to happen. I have those changes ready along with updated merge tests, so I'll publish that so folks can take a look

Reply:
Makes sense to me.